### DISCUSSION TITLE
How to improve the performance of ByteTrack

### DISCUSSION THREAD

**Discussion by westlinkin** [roboflow/supervision#1001] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

I am tracking multiple objects ( more than 20 objects), I've adjust the parameters of `ByteTrack` to:
```
byte_tracker = sv.ByteTrack(track_buffer=30,
                            track_thresh=0.35,
                            match_thresh=.95,
                            frame_rate=59)
```

first of all, I found that `track_buffer` does not have any effect on the result. when increase `match_thresh`, the tracking is more 'stable', at the value of `0.95`, even when the objects collide, the tracking is still working (with the same track id). 

But one object is not tracking well, the track id switch to a new one even though there is no other object nearby, which is quite confusing.  as the video shows, tracker id 15  becomes 30 for no reason (in my opinion), How to avoid that?

https://github.com/roboflow/supervision/assets/1687198/be79e242-8af0-4eee-9512-84de390ed1ce

**Comment by SkalskiP** [comment]
Hi @westlinkin! 👋🏻 LEt me convert that issue into a discussion and put it into the Q&A section.

**Comment by SkalskiP** [comment]
@westlinkin tracker ID 15 becomes 30 because you get double detection for a split second. The arm pad of the player gets misclassified as head. And that's why a new tracker gets spawned.

![Screenshot 2024-03-15 at 10 56 06](https://github.com/roboflow/supervision/assets/26109316/c9431a10-1b90-40c7-98a3-0f99f1824fcc)

The solution for that would be to have a way to suppress the assignment of a new ID until the object is visible for multiple subsequent frames. I was under the impression that it's the role of `track_buffer`, but you say it does not work.

**Comment by westlinkin** [comment]
Thanks! I set the track_buffer to 3, 30, 3000, this always happens, so I think `track_buffer` does not work. 

After I filter the detections with confidence > 0.5, this problem goes away. but there are a lot of frames with no certain track id (because they got filtered). Maybe there should be a Tool to 'fill in' the detections (sort like smoother)
